src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs

```diff
@@ -381,6 +383,135 @@ public Task GetMigratedServerUrls()
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    RedirectStandardInput = true,
+                    UseShellExecute = false
+                };
+
+                Process process = null;
+                try
+                {
+                    process = Process.Start(processStartInfo);
+                }
+                catch (Exception e)
+                {
+                    process?.Kill();
+                    throw new InvalidOperationException("Unable to execute Migrator." + Environment.NewLine +
+                                                        "Command was: " + Environment.NewLine +
+                                                        (processStartInfo.WorkingDirectory ?? Directory.GetCurrentDirectory()) + "> "
+                                                        + processStartInfo.FileName + " " + processStartInfo.Arguments, e);
+                }
+
+                await process.StandardInput.WriteLineAsync(migrationConfiguration.Input.ToString());
+
+                if (migrationConfiguration.IsExportCommand == false)
+                {
+                    try
+                    {
+                        var line = process.StandardOutput.ReadLine();
+                        // return type is json, will throw if we fail to deserialize
+                        JsonConvert.DeserializeObject(line);
+                        using (var sw = new StreamWriter(ResponseBodyStream()))
+                        {
+                            await sw.WriteAsync(line);
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        var error = process.StandardError.ReadLine();
+                        process.Kill();
+                        throw new InvalidOperationException($"Process error: {error}, exception: {e}");
+                    }
+
+                    return;
+                }
+
+                var operationId = GetLongQueryString("operationId", false) ?? Database.Operations.GetNextOperationId();
+                var token = CreateOperationToken();
+                var transformScript = migrationConfiguration.TransformScript;
+
+                var t = Database.Operations.AddOperation(Database, $"Migration from: {migrationConfiguration.DatabaseTypeName}",
+                    Operations.OperationType.DatabaseMigration,
+                    onProgress =>
+                    {
+                        return Task.Run(() =>
+                        {
+                            var result = new SmugglerResult();
+
+                            try
+                            {
+                                using (ContextPool.AllocateOperationContext(out DocumentsOperationContext migrateContext))
+                                using (var stream = new GZipStream(process.StandardOutput.BaseStream, CompressionMode.Decompress))
```

Comment:
You are sending GZIP over the pipe? Why?
It is in process communication and the cost of compression is only wasting our time.

Reply:
done